Sources/Amplitude/Amplitude.m

```diff
@@ -753,12 +759,13 @@ - (void)uploadEventsWithLimit:(int)limit {
         
         long identifyCount = [self->_identifyBuffer count];
         long numIdentify = limit > 0 ? fminl(identifyCount, limit) : identifyCount;
-        
         NSMutableArray *events = [[self->_eventsBuffer subarrayWithRange:NSMakeRange(0, numEvents)] mutableCopy];
         NSMutableArray *identifys = [[self->_identifyBuffer subarrayWithRange:NSMakeRange(0, numIdentify)] mutableCopy];
-        NSDictionary *merged = [self mergeEventsAndIdentifys:events identifys:identifys numEvents:(numEvents+numIdentify)];
-
         for (NSDictionary *event in events) {
+            long long currentSequenceNumber = [[event objectForKey:@"sequence_number"] longLongValue];
+            if (currentSequenceNumber <= self->_maxEventSequenceNumber) {
```

Comment:
This is for fixing the bug found from testLogEventUploadLogic test. 
If the response is 200 but didn't succeed, we won't do anything to the file storage and buffers. So the second try will restore all events in the buffer to the file storage which leading file storage have duplicates events.

Reply:
When will it happen for the failure with 200 status code?

Reply:
Before the changes, retry will re-upload the events in the event buffer into file storage. So there will have duplicate events in file storage.  
This is the reason why I add those two properties. It will skip the event which already exists in the file storage. 
@property (nonatomic, assign) long long maxEventSequenceNumber;
@property (nonatomic, assign) long long maxIdentifySequenceNumber;

Reply:
Without the logic testLogEventUploadLogic will fail, the `queuedEventCount` is double the `event buffer` count because we re-store the event into the file storage after failure with 200.

Without exposing updatingCurrently, how can we test this issue out?